framework/src/mfem/problem_operators/ProblemOperatorInterface.C

```diff
@@ -48,7 +51,13 @@ ProblemOperatorInterface::SetTestVariablesFromTrueVectors()
 {
   for (unsigned int ind = 0; ind < _test_variables.size(); ++ind)
   {
-    _test_variables.at(ind)->SetFromTrueVector();
+    auto * const test_var = _test_variables.at(ind);
+
+    // We must sync the memory flags from the true true vector to the grid function copy of the true
+    // vector
+    mooseAssert(_test_true_vector, "The true vector should already have been set");
+    test_var->GetTrueVector().SyncMemory(*_test_true_vector);
```

Comment:
I'm not sure I understand this - `X` is the vector storing all (true) grid function DoFs. If `_test_variables.size()` is greater than 1, `_test_true_vector` will be longer than the last `test_var` true DoF vector

Reply:
It is a little silly/rendudant to do this for all `_test_variables`. We should just do this for the zeroth test variable

Reply:
I'm not sure what the `Registered` flag is for in MFEM, but I do know that `flags & Registered` is `true` so we never do the check in the `MFEM_ASSERT`

Reply:
Understood - this makes sense now with the usage of `_test_true_vector` formed in `Init`. I agree each grid function needs its own flags synced with X.